scripts/lint.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -eou pipefail
+
+# relative path calculation shim, I hope python is installed!
+function relative_path {
+    if hash realpath 2> /dev/null; then
+        realpath --relative-to="$2" "$1"
+    else
+        if hash python 2> /dev/null; then  
+            python -c "import os.path; print os.path.relpath('$1', '$2')"
+        else
+            echo "unable to calculate a relative path"
+            exit 1
+        fi
+    fi    
+}
+
+# This is all in a single file for simplicity of future updates
+
+gometalinter_file="$(mktemp)"
+errcheck_excludes_file="$(mktemp)"
+
+# TODO: download the metalinter config and base errcheck file from a shared repo?
```

Comment:
all of the scripts in `./scripts` could come from the same submodule tbh

Reply:
Yeah, submodules are kind of a pain though, and still require each repository to opt in to updates (which may or may not be desirable).  I'm also not happy with a file that downloads off the internet to execute, so I think we should probably spend some more time discussing the longer term sharing strategy for these things, maybe we settle on submodule or something managed by the Makefile? I don't know...

Reply:
I don't disagree, but OTOH duplicating these files in each repository is kinda overkill. We could also vendor them (from Core) I guess - but that doesn't feel like a great option either?